### PR TITLE
feat: add clear function

### PR DESCRIPTION
**Pull request by rfyiamcool**
### change

add clear() function.

https://github.com/dolthub/swiss/blob/main/map.go#L238

### CODE REVIEW

File: concurrent_swiss_map.go [hunk 1]
```diff
@@ -95,6 +95,16 @@ func (m *CsMap[K, V]) Has(key K) bool {
 	return shard.items.HasWithHash(key, hashShardPair.hash)
 }
 
+func (m *CsMap[K, V]) Clear() {
+	for i := 0; i < len(m.shards); i++ {
+		shard := m.shards[i]
+
+		shard.RLock()
```
Comment (mhmtszr) [inline concurrent_swiss_map.go:102]: we shouldn't be able to update the map while cleaning, we need to use write lock here what do you think?

**Review by mhmtszr — CHANGES_REQUESTED** [review]

File: concurrent_swiss_map.go (hunk 1 above)
Comment (rfyiamcool) [reply concurrent_swiss_map.go]: 😁 you are right, I modified this code based on range, this is my mistake.

**Comment by mhmtszr** [comment]
@rfyiamcool I think we should also clear groups what do you think? keys and values are stored in groups after the clear function garbage collection should be able to clears the memory corresponding to the map's heap.

**Comment by rfyiamcool** [comment]
> @rfyiamcool I think we should also clear groups what do you think? keys and values are stored in groups after the clear function garbage collection should be able to clear the memory corresponding to the map's heap.

Is that so? 😁

**Comment by mhmtszr** [comment]
@rfyiamcool thanks for the contribution.

**Pull request merged**